crates/router/src/analytics.rs

```diff
@@ -653,11 +653,42 @@ pub mod routes {
             state.clone(),
             &req,
             json_payload.into_inner(),
-            |state, auth: AuthenticationData, req, _| async move {
+            |state, auth: UserFromToken, req, _| async move {
+                let role_id = auth.role_id;
+                let role_info =
+                    RoleInfo::from_role_id(&state, &role_id, &auth.merchant_id, &auth.org_id)
+                        .await
+                        .change_context(UserErrors::InternalServerError)
+                        .change_context(OpenSearchError::UnknownError)?;
+                let permissions = role_info.get_permissions_set();
+                let accessible_indexes: Vec<_> = vec![
+                    (
+                        SearchIndex::PaymentAttempts,
+                        vec![Permission::PaymentRead, Permission::PaymentWrite],
+                    ),
+                    (
+                        SearchIndex::PaymentIntents,
+                        vec![Permission::PaymentRead, Permission::PaymentWrite],
+                    ),
+                    (
+                        SearchIndex::Refunds,
+                        vec![Permission::RefundRead, Permission::RefundWrite],
+                    ),
+                    (
+                        SearchIndex::Disputes,
+                        vec![Permission::DisputeRead, Permission::DisputeWrite],
+                    ),
+                ]
+                .into_iter()
+                .filter(|(_, perm)| perm.iter().any(|p| permissions.contains(p)))
+                .map(|i| i.0)
+                .collect();
+
                 analytics::search::msearch_results(
                     &state.opensearch_client,
                     req,
-                    &auth.merchant_account.merchant_id,
+                    &auth.merchant_id,
+                    accessible_indexes,
```

Comment:
what happens when user has neither of the permissions? accessible_indexes = vec![]?, will msearch run on this?

Reply:
Resolved this by changing the `OpenMserchOutput` struct to handle cases when the user has none of the permissions to access indexes.

Reply:
can you share the the output of this test case

Reply:
![Screenshot 2024-06-26 at 1 50 44 PM](https://github.com/juspay/hyperswitch/assets/83278309/d1a93ab5-aeac-4f03-ad55-dbba360d7b3f)
<img width="589" alt="Screenshot 2024-06-26 at 1 50 08 PM" src="https://github.com/juspay/hyperswitch/assets/83278309/73e7044e-a1a4-4978-829d-e2d71e0fd794">